### ISSUE TITLE
Dory seems broken on Mint

### ISSUE THREAD

**Issue by rmNyro** [FreedomBen/dory#13]
Hi again,

I completely lost so I'm quickly posting to tell what's what's but I might be too much concise.
I'm testing dory on `Linux Mint 18.1 Serena` as you now know. I have no container (before running any test obviously) running except for portainer and dory:
![dory status](https://user-images.githubusercontent.com/1544703/29824364-aaa8b4ce-8cd1-11e7-81d9-2f93dff63e68.png)

The first strange thing is that the domain name doesn't really matter.
i.e.: `docker run -it -e VIRTUAL_HOST=s -e VIRTUAL_PORT=8000 crccheck/hello-world` can be accessed via `http://fgsdg.docker/` or anything ending with `.docker`.
I suppose it is not exactly the desired behavior (or is it?).

The second problem is that only DN ending by `.docker` are accessible from the browser. If another server has another custom fancy DN, it's inaccessible (the first launched gets the response) because of that.
So to access it, you have to specify a "correct" DN ending with `.docker`.

Sooooo, is my Mint a hater to dory? I have no background in ruby and very little in DevOps/networking so I'm afraid I can't quite help but to report. :/

And again, I hope it helped. :)

**Comment by FreedomBen** [comment]
Yep, the warning are not related (and I'll try to fix those soon cause they're annoying).

I have seen this before, but it's really hard to reproduce.  It seems like the docker networking and the proxy container get into some weird state where they don't behave properly.  Each time one of the following seems to fix it:

1.  Restart docker (this will kill dory's docker containers, so need to start dory again after):
```
    sudo systemctl restart docker
    dory up
```
2.  Double check `VIRTUAL_HOST` variables and `VIRTUAL_PORT`.  Sometimes if the downstream container isn't reachable the nginx proxy will pick one as a "default."  The unreachability can be caused by little typos in host or port.

3.  Attach to the proxy, then restart your application containers and make sure you see the output:

```dory attach proxy  # this will take over the current terminal, so run next commands in separate terminal windows
docker-compose down
docker-compose up```
